Add Corsika7ID (+converter) and Particle.from_nucleus() (#426)
scikit-hep/particle: README.rst

    >>>
    >>> Particle.from_evtgen_name("J/psi")
    <Particle: name="J/psi(1S)", pdgid=443, mass=3096.900 ± 0.006 MeV>
    >>>

>     >>> Particle.from_nucleus_info(a=12, z=6)

Reviewer: I understand your point about a method special for nuclei. I can see the interest, especially if we want to expand the relevance of the package to other colleagues.

Just note for completeness that you can also do

```
>>> Particle.findall(lambda p: p.pdgid.A == 12 and p.pdgid.Z == 6)
[<Particle: name="C12", pdgid=1000060120, mass=11177.9291399 MeV>]
```

which is as here but a list, though, since using one of the find methods.

Author: I didn't even know that PDGID has `.A` and `.Z`, nice!

Reviewer: You can see *a lot" in the notebook demo but point taken that we need to invest on a Sphinx doc. The fact that you did not see it should push this up in priority ... Of course any help is welcome :-).

.. code-block:: python

    >>> from particle import Particle

>     >>> from particle import Geant3ID, PythiaID, Corsika7ID

Reviewer: ```suggestion
    >>> from particle import Corsika7ID, Geant3ID, PythiaID
```

    >>> p.name
    'pi+'

    >>> cid = Corsika7ID(5)
    >>> p = Particle.from_pdgid(cid.to_pdgid())
    >>> p.name
    'mu+'

> Corsika7

Reviewer: I see that you even add a little section in the README for Corsika7. We do not have it for others but I see it is a good thing. You're telling me we could/should expand this doc section on converters (at least while we do not have Sphinx docs). OK, cool.

^^^^^^^^

> The ``Corsika7ID`` class implements some features to make it easier to work with Corsika7 output.

Reviewer: ```suggestion
The ``Corsika7ID`` class implements features to make it easier to work with Corsika7 output.
```

> For a full featureset, look into the ``particle.corsika`` module.

Reviewer: ```suggestion
For a full feature set, please refer to the ``particle.corsika`` submodule.
```

``Corsika7ID.from_particle_description(from_particle_description: int)`` returns ``(Corsika7ID, bool)``
to automatically parse the ``particle_description`` from the Corsika7 particle data sub-block.

> ``Corsika7ID.is_particle()`` checks if the id refers to an actual particle or something else (like additional information).

Reviewer: ```suggestion
``Corsika7ID.is_particle()`` checks if the ID refers to an actual particle or something else (like additional information).
```
